### DISCUSSION TITLE
Passkeys created in Librewolf not recognized by Firefox

### DISCUSSION THREAD

**Discussion by debmint** [keepassxreboot/keepassxc#11080] [Q&A]
I've set up some passkeys using Librewolf.  They work fine under Librewolf and Brave, but they don't work in Firefox.  I just get a prompt to enable my device and it times out if I wait.  I checked to make sure Passkeys are enable in Firefox, and they are.  When I am on the login page, on Firefox and all the others the page says the browser has partial support for passkeys.

Would anyone have any idea what else I might look for to see why this is occurring?

**Comment by varjolintu** [comment]
Does `about:config` show any differences between `webauthn` and `passkey` settings when compared to Librewolf?

**Reply by debmint** [reply]
These options are identical.  As far as I can see, Librewolf has several settings that are stricter than Firefox, FWIW, in general, the extension works in Firefox, it fills in username and passwords correctly, and everything except passkeys.  TOTP work in Firefox.  I may try creating a passkey with Firefox and see how that works.

Thanks for the input.

**Reply by droidmonkey** [reply]
What version of Firefox are you running? They have been adding passkey features over time and you may have a drastically older version than Librewolf is at. This can happen if you are using the ESR version of Firefox.

**Reply by ghost** [reply]
They are both 128.0 (Debian)

**Reply by debmint** [reply]
> They are both 128.0 (Debian)

After a month letting it slide, I revisited the issue, and I found that Passkeys were not enabled in Firefox.  I could have sworn I checked that at first!

Now it seems that If I try to log in to GitLAB, it pauses on prompt to touch device.  If I cancel and retry, it works.  I saw on discussion in maybe keepassxc-browser repo a few months ago that gitlab was not yet functional.  One odd thing, GitLab passkeys are stored under "KeePassXC-Browser-Passwords", and GitHub passkeys are stored under "Imported Passkeys".  I wonder what is the difference?

**Reply by varjolintu** [reply]
> > They are both 128.0 (Debian)
> 
> After a month letting it slide, I revisited the issue, and I found that Passkeys were not enabled in Firefox. I could have sworn I checked that at first!
> 
> Now it seems that If I try to log in to GitLAB, it pauses on prompt to touch device. If I cancel and retry, it works. I saw on discussion in maybe keepassxc-browser repo a few months ago that gitlab was not yet functional. One odd thing, GitLab passkeys are stored under "KeePassXC-Browser-Passwords", and GitHub passkeys are stored under "Imported Passkeys". I wonder what is the difference?

"Imported Passkeys" is used by default when importing passkeys.

**Reply by debmint** [reply]
> > After a month letting it slide, I revisited the issue, and I found that Passkeys were not enabled in Firefox. I could have sworn I checked that at first!
> > One odd thing, GitLab passkeys are stored under "KeePassXC-Browser-Passwords", and GitHub passkeys are stored under "Imported Passkeys". I wonder what is the difference?

> "Imported Passkeys" is used by default when importing passkeys.

I cannot understand why the difference in storage.  github is stored diretly in the "Imported Passkeys" folder, while gitlab is stored in "KeePassXC-Browser-Passwords" subfolder of "Imported Passkeys"

**Reply by varjolintu** [reply]
> I cannot understand why the difference in storage. github is stored diretly in the "Imported Passkeys" folder, while gitlab is stored in "KeePassXC-Browser-Passwords" subfolder of "Imported Passkeys"

Is it possible that you've accidentally arranged the groups? You can always move the credentials around.

**Comment by debmint** [comment]
On Thu, 2024-08-15 at 20:36 -0700, Sami Vänttinen wrote:
> > Now it seems that If I try to log in to GitLAB, it pauses on prompt
> > to touch device. If I cancel and retry, it works. I saw on
> > discussion in maybe keepassxc-browser repo a few months ago that
> > gitlab was not yet functional. One odd thing, GitLab passkeys are
> > stored under "KeePassXC-Browser-Passwords", and GitHub passkeys are
> > stored under "Imported Passkeys". I wonder what is the difference?
> "Imported Passkeys" is used by default when importing passkeys.

I wonder if the fact that GitLab's passkey is stored differently is due
to the fact that they possibly use some different protocol or
something?

**Reply by varjolintu** [reply]
GitLab has some problems during the passkeys development. I need to recheck to current situation.

**Comment by debmint** [comment]
On Fri, 2024-08-16 at 22:34 -0700, Sami Vänttinen wrote:
> GitLab has some problems during the passkeys development. I need to
> recheck to current situation.

As I mentioned in a previous post, it hangs, so to speak, on first
challenge, but it appears that if you retry, it succeeds.

**Reply by varjolintu** [reply]
It seems that there might be a script at the page that tries to authenticate with a passkey immediately before the page content (or content script) is loaded, causing the extension not to detect the API call.

**Comment by debmint** [comment]
On Mon, 2024-08-19 at 20:38 -0700, Sami Vänttinen wrote:
> It seems that there might be a script at the page that tries to
> authenticate with a passkey immediately before the page content (or
> content script) is loaded, causing the extension not to detect the
> API call
Thank you for the reply.  Your answer agrees with the suspicion I had
come up with.

**Comment by debmint** [comment]
On Tue, 2024-09-03 at 21:10 -0700, Sami Vänttinen wrote:
> > I cannot understand why the difference in storage. github is stored
> > diretly in the "Imported Passkeys" folder, while gitlab is stored
> > in "KeePassXC-Browser-Passwords" subfolder of "Imported Passkeys"
> Is it possible that you've accidentally arranged the groups? You can
> always move the credentials around.
No, I don't think so.  Github went directly into "Imported Passkeys",
and, as a matter of fact, I deleted my gitlab passkey and created a new
one, and, just like before, it went into the "KeePassXC-Browser
Passwords" subgroup of "imported...".   I have another account which
went in with gitlab, also.  Oh, well, just as long as they work ...

**Comment by ieugen** [comment]
I have used keepassxc with Firefox and I created a passkey for Kanidm auth server (homelab). 
The passkeys work in Firefox but they don't work in LibreWolf. 
They are not detected. 

This is what I get when I press Passkey button. 
![keepassxc-missing-passkey](https://github.com/user-attachments/assets/d5f382d9-6d74-438d-8360-f8b62e6288c0)

It does work with Firefox ESR - 128 I think and Firefox 137b .
It does not work with Librewolf 136. 
These are my browser options. 

![image](https://github.com/user-attachments/assets/70d17b7c-40d5-4dd6-b093-d9caad4fb8c4)

I'm adding info here in the hope it's helpful. From the facts, it seems like it's a LibreWolf issue.

**Reply by varjolintu** [reply]
Make sure you have enabled the option in the extension settings?

**Reply by ieugen** [reply]
Thanks  @varjolintu . 

I have enabled passkeys in the extension and now it works. 
It did not seem to work in private mode, but i guess it's because I just enabled it. 

![image](https://github.com/user-attachments/assets/f1860d90-51a4-4c53-940f-1b26340a2097)
